### PR TITLE
7217 error handling

### PR DESCRIPTION
**Pull request by gdicristofaro**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java [hunk 1]
```diff
@@ -347,29 +441,43 @@ private void checkAttributeType(Collection<BlackboardAttribute> bbattributes, St
...
             columnPosition++;
         }
 
+        if (columnsToProcess.size() != attrList.size()) {
+            String missingColumns = IntStream.range(0, attrList.size())
+                    .filter((idx) -> !columnsToProcess.containsKey(attrList.get(idx).getAttributeName()))
+                    .mapToObj((idx) -> String.format("'%s'", attrList.get(idx).getColumnName() == null ? "<null>" : attrList.get(idx).getColumnName()))
+                    .collect(Collectors.joining(", "));
+
+            logger.log(Level.SEVERE, String.format("Columns size expected not found in file %s based on xml from %s. Column Keys Missing = [%s]; Header Line = '%s'.",
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:474]: I'm not sure if this is severe. I think a warning would be better.

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java [hunk 2]
```diff
@@ -437,29 +557,66 @@ private void getArtifactNode(Document xmlinput) {
...
+                    logger.log(Level.SEVERE, String.format("No known attribute mapping found for [%s]", getXmlAttrIdentifier(parentName, attributeName)));
+                }
+
+                if (required != null && required.compareToIgnoreCase("yes") != 0 && required.compareToIgnoreCase("no") != 0) {
+                    logger.log(Level.SEVERE, String.format("Required value %s did not match 'yes' or 'no' for [%s]",
+                            required, getXmlAttrIdentifier(parentName, attributeName)));
+                }
+
+                if (columnName == null) {
+                    logger.log(Level.SEVERE, String.format("No column name provided for [%s]", getXmlAttrIdentifier(parentName, attributeName)));
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:601]: I'm not sure these 3 log messages are severe. I think a warning would be better.

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java [hunk 3]
```diff
@@ -258,16 +320,22 @@ private void processFile(File LeappFile, List<List<String>> attrList, String fil
...
+     * @param fileName name of file begin processed
      *
      * @return
      */
     private Collection<BlackboardAttribute> processReadLine(String line, Map<Integer, String> columnNumberToProcess, String fileName) throws IngestModuleException {
-         
+        if (MapUtils.isEmpty(columnNumberToProcess)) {
+            return Collections.emptyList();
+        } else if (line == null) {
+            logger.log(Level.SEVERE, "Line is null.  Returning empty list for attributes.");
```
[the remark this answers is not in the document]
Comment (gdicristofaro) [reply Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:333]: Sure.  Will do.

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java [hunk 4]
```diff
@@ -228,26 +282,34 @@ private void processLeappFiles(List<String> LeappFilesToProcess, Content dataSou
 
     }
 
-    private void processFile(File LeappFile, List<List<String>> attrList, String fileName, BlackboardArtifact.Type artifactType,
+    private void processFile(File LeappFile, List<TsvColumn> attrList, String fileName, BlackboardArtifact.Type artifactType,
             List<BlackboardArtifact> bbartifacts, Content dataSource) throws FileNotFoundException, IOException, IngestModuleException,
             TskCoreException {
+
+        if (LeappFile == null || !LeappFile.exists() || fileName == null) {
+            logger.log(Level.SEVERE, String.format("Leap file: %s is null or does not exist", LeappFile == null ? LeappFile.toString() : "<null>"));
```
[the remark this answers is not in the document]
Comment (gdicristofaro) [reply Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:290]: I believe the code does a scan of the tsv files present in the ModuleOutput xLEAPP directory and, for each tsv file found that should be parsed, calls this method.  If we hit this, I think it means something went quite wrong.

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java [hunk 5]
```diff
@@ -228,26 +282,34 @@ private void processLeappFiles(List<String> LeappFilesToProcess, Content dataSou
...
+            return;
+        } else if (attrList == null || artifactType == null || dataSource == null) {
+            logger.log(Level.SEVERE, String.format("attribute list, artifact type or dataSource not provided for %s", LeappFile == null ? LeappFile.toString() : "<null>"));
```
[the remark this answers is not in the document]
Comment (gdicristofaro) [reply Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:293]: Same here.  If we are in this method, we should have a valid list of attributes, an artifact type, and some data source that we ran xLEAPP on.

File: Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java (hunk 2 above)
Comment (gdicristofaro) [reply Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:601]: If this flag gets hit, I think that identifies malformed xml.

Comment (dannysmyda) [reply Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java:601]: Okay, sounds good.

**Review by dannysmyda — APPROVED** [review]

**Review by markmckinnon — APPROVED** [review]

**Pull request merged**